### ISSUE TITLE
Local Optimizer

### ISSUE THREAD

**Issue by tmulc18** [tmulc18/Distributed-TensorFlow-Guide#6]
There is an issue with using adagrad instead of sgd on the worker level... the second worker never starts.

**Comment by AshAswin** [comment]
Thanks for your fix! Even, I faced a similar issue with optimizers other than Gradient Descent. I would like to get more information on this issue
